core-common/src/main/java/org/mapstruct/util/ComponentModelMapper.java

```diff
@@ -0,0 +1,44 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.util;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotations that are marked with this annotation are used to set the component model like {@code spring}.<br>
+ *
+ * <p>It is never allowed to have more than one component model annotation at one
+ * {@link org.mapstruct.Mapper @Mapper}.<br>
+ * In case there is no annotation of this kind the {@code default} component model will be used.
+ *
+ * @author Christian Bandowski
+ */
+@Documented
+@Target(ElementType.ANNOTATION_TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ComponentModelMapper {
```

Comment:
I had to add this meta-annotation that allows to recognize the `@SpringMapper`/.. annotations, otherwise it is not possible to detect that no such annotation exist and the `default` component model should be used - or if multiple component model annotations exists (-> ERROR).